src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionReader.cs

```diff
@@ -607,6 +605,10 @@ private CollectionSetItemDelegate GetCollectionSetItemDelegate<T>(CollectionData
             }
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2060:MakeGenericMethod",
+                Justification = "The call to MakeGenericMethod is safe due to the fact that ObjectToKeyValuePairGetKey and ObjectToKeyValuePairGetValue are not annotated.")]
+        private MethodInfo MakeGenericMethod(MethodInfo method, Type keyType, Type valueType) => method.MakeGenericMethod(keyType, valueType);
```

Comment:
Maybe put this as a nested function inside `GetCollectionSetItemDelegate`, that way only that method can call it and we can assume which `MethodInfo`s are passed into it.

Reply:
Repeated elsewhere.